src/main/java/com/shelterhub/service/AnimalService.java

```diff
@@ -0,0 +1,58 @@
+package com.shelterhub.service;
+
+import com.shelterhub.database.AnimalRepository;
+import com.shelterhub.domain.model.Animal;
+import com.shelterhub.dto.AnimalDTO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Service
+public class AnimalService {
+
+    @Autowired
+    private AnimalRepository animalRepository;
+
+    public AnimalDTO create(AnimalDTO animalDTO) {
+        var animalToBePersisted = AnimalDTO.toAnimal(animalDTO);
+        var createdAnimal = animalRepository.save(animalToBePersisted);
+
+        return Animal.toDTO(createdAnimal);
+    }
+
+    public AnimalDTO update(AnimalDTO animalDTO, UUID animalId) {
+        Animal animal = animalRepository.getReferenceById(animalId);
+
+        Animal animalToBePersisted = AnimalDTO.toAnimalWithSameAnimalId(animalDTO, animalId);
```

Comment:
como aqui a gnt não tá usando nada do animal buscado, poderia ser:
```suggestion
        Animal animalToBePersisted = animalDTO.toAnimal();
```